### PR TITLE
CRI: remove PodSandboxStatus.Linux.Namespaces.Network

### LINKED ISSUE #44972: CRI: Why do we need PodSandboxStatus.Linux.Namespaces.Network?
Now, in CRI we expose the network namespace in `PodSandboxStatus`, but kubelet is not using it.

Actually, now only network plugin needs the network namespace, but the network logic is moved down to runtime.

There seems to be no reason to keep this field except for debugging

@yujuhong @feiskyer 
@kubernetes/sig-node-api-reviews

### PR DESCRIPTION
**Pull request by feiskyer**
**What this PR does / why we need it**:

PodSandboxStatus.Linux.Namespaces.Network is not used, so this PR removes it from CRI.

**Which issue this PR fixes**

Closes: #44972

**Special notes for your reviewer**:

**Release note**:

```release-note
Remove PodSandboxStatus.Linux.Namespaces.Network from CRI.
```

/assign @Random-Liu @yujuhong

### CODE REVIEW

**Comment by feiskyer** [comment]
@k8s-bot verify test this

**Comment by feiskyer** [comment]
Rebased.

File: pkg/kubelet/api/v1alpha1/runtime/api.proto [hunk 1]
```diff
@@ -338,8 +338,6 @@ message PodSandboxNetworkStatus {
 
 // Namespace contains paths to the namespaces.
 message Namespace {
-    // Path to the network namespace.
-    string network = 1;
     // Namespace options for Linux namespaces.
     NamespaceOption options = 2;
```
Comment (xiangpengzhao) [inline pkg/kubelet/api/v1alpha1/runtime/api.proto]: We can't change this number to 1, right?

**Pull request merged**